kriskuiper/cmd-digital-playground
# Feat/storyblok preview

## Description
I found a (very nasty) way to implement the live updating preview. What happens is that we check if the website is in preview context by using the [Storyblok JS Bridge](https://www.storyblok.com/docs/Guides/storyblok-latest-js). If so, we hijack the page with JavaScript and remove all child elements from the `modular-content` component, get new data and inject the updated components with JavaScript... 😐.

This means we now have to build a component two times 🤷‍♂️: once in HTML for the static site generation and once in `site/assets/js/preview/components` as a JS function to make it update in the preview. I definitely agree that this is not ideal so we have to find a nicer way in the future but for now it works.

## Testing
To test this PR you have to check out to the branch and do the following:
1. Spin up a development environment by running `npm run dev`
2. Open up storyblok and go to the home page
3. Add or update a component and click save or do `cmd + s`

You could also do the following:
1. Go to https://app.storyblok.com
2. Enter `cmd-playground` workspace and click settings
3. Change location (default environment) to https://cmd-digital-playground-git-feat-storyblok-preview.kriskuiper.now.sh
4. Open the home page in storyblok
5. Add or update a component and click save or do `cmd + s`

> The preview should now update 💖

## To do after merging this PR
I'll create some tickets to do some follow up:
- [ ] Make clicking on a component open the editor sidebar
- [ ] Make the preview update in a nicer way
